portfolio/src/main/webapp/script.js

```diff
@@ -7,7 +7,7 @@ function getRandomQuote(){
 
 /** Fetch comments from server and add to DOM */
 function getComments(){
-    fetch('/data').then(response => response.json()).then((messages) => {
+    fetch('/data?num='+document.getElementById("num-comments").value).then(response => response.json()).then((messages) => {
```

Comment:
I would take the document.getElementById().value part and make that a variable before this line get called, or at least a function inlined.

const numComments = ...
fetch('/data?num=' + numComments)...

Or

fetch('/data?num=' + getNumComments())...